drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/referral_postings_loader.rb

```diff
@@ -303,24 +309,40 @@ def build_posting_records
           log_info "#{row.context} Skipping Referral #{referral_id} - no HoH"
           next
         end
-        hoh_mci_id = row_value(hoh_member_row, field: 'MCI_ID')
-        # 2) find the household_id for the project enrollment with that MCI
-        household_id = referral_household_id(hoh_mci_id, project_id)
-        # Household must be present for Accepted and Accepted Pending postings
+
         status = posting_status(row)
+        # 2) Find the household_id for the enrollment linked to this ReferralPosting row, if any.
+        household_id = nil
+        if status == accepted_status
+          # For Accepted referrals, the Enrollment ID should be present on the row.
+          enrollment_id = row_value(row, field: 'ENROLLMENTID', required: false)
+          unless enrollment_id.present?
+            log_info("#{posting_row.context} ENROLLMENTID missing from Accepted referral")
+            next
+          end
+          household_id = household_id_by_enrollment_id[enrollment_id]
+        elsif status == accepted_pending_status
+          # For Accepted Pending referrals, we should have created the WIP Enrollment(s) in an earlier step.
+          # find the HoH's MCI ID
+          hoh_mci_id = row_value(hoh_member_row, field: 'MCI_ID')
+          # find the household_id for the project enrollment with that MCI
+          household_id = referral_household_id(hoh_mci_id, project_id)
+        end
```

Comment:
Not sure how I missed this one before! For Accepted referrals, we should use the `ENROLLMENTID` to determine the household_id that's linked to the Posting. We were getting several hundred logs where Accepted postings were skipped because we couldn't tie them to an existing household, and I think this is why.

Reply:
Deploying & testing this code now.

Reply:
Nice! That makes sense, sorry I missed that

Reply:
All good!